modules/renter/hostdb/hostdb.go

```diff
@@ -107,11 +108,13 @@ func newHostDB(cs consensusSet, d dialer, s sleeper, p persister, l *persist.Log
 		log:     l,
 
 		// TODO: should index by pubkey, not ip
-		activeHosts: make(map[modules.NetAddress]*hostNode),
+		activeHosts: make(map[modules.NetAddress]*hostEntry),
 		allHosts:    make(map[modules.NetAddress]*hostEntry),
 		scanPool:    make(chan *hostEntry, scanPoolSize),
 	}
 
+	hdb.hostTree = hosttree.New(hdb.calculateHostWeight())
```

Comment:
@DavidVorick is right -- you can just pass `hdb.calculateHostWeight` if you modify it to satisfy the `weightFunc` type. It should be:

```go
func (hdb *HostDB) calculateHostWeight(entry modules.HostDBEntry) types.Currency {
    // use hdb.blockheight internally
}
```